core/src/main/scala/cats/std/try.scala

```diff
@@ -78,6 +97,14 @@ trait TryInstances extends TryInstances1 {
     }
 }
 
+private[std] object TryInstances {
+  /**
+   * A `Failure` can be statically typed as `Try[A]` for all `A`, because it
+   * does not actually contain an `A` value (as `Success[A]` does).
+   */
+  def castFailure[A](f: Failure[_]): Try[A] = f.asInstanceOf[Try[A]]
```

Comment:
worth it to put an `@inline` here? I don't actually know how effective that is, but it seems like a candidate.